qprofiler2/src/org/qcmg/qprofiler2/summarise/KmersSummary.java

```diff
@@ -97,7 +106,11 @@ public static String producer(final int k, final String mers , final boolean inc
 		String str1 = producer( k, "", includeN );			
 		return str1.split(Constants.COMMA_STRING);
 	}	
-	public void parseKmers( byte[] readString, boolean reverse , int flagFirstOfPair){
+	public void parseKmers( byte[] readString, boolean reverse , int flagFirstOfPair) throws Exception{
+		 if( readString.length > maxCycleNo ) {
+			 throw new Exception("Can't handle large read sequence, which length is greater than " + maxCycleNo + "!");
```

Comment:
Don't throw an Exception, use one of its subclasses.
IllegalArgumentException would be appropriate here

Reply:
updated, good suggestion